controllers/setup.go

```diff
@@ -35,11 +44,27 @@ func SetupControllers(mgr manager.Manager, supportExtendedDaemonset bool) error
 		Scheme:      mgr.GetScheme(),
 		Recorder:    mgr.GetEventRecorderFor("DatadogAgent"),
 		Options: datadogagent.ReconcilerOptions{
-			SupportExtendedDaemonset: supportExtendedDaemonset,
+			SupportExtendedDaemonset: options.SupportExtendedDaemonset,
 		},
 	}).SetupWithManager(mgr); err != nil {
 		return fmt.Errorf("unable to create controller DatadogAgent: %w", err)
 	}
 
+	ddClient, err := datadogclient.InitDatadogClient(options.APIKey, options.AppKey)
+	if err != nil {
+		return fmt.Errorf("unable to create Datadog API Client: %w", err)
+	}
```

Comment:
does it means that api and app key are mandatory to start the operator?

Reply:
we can work on it in another PR.

Reply:
yes, at the moment. open to other ideas since it will be a breaking change

Reply:
until we don't release an new operator with this change it is only a work in progress :)

Reply:
I think it should lazily instantiated and/or be a singleton

Reply:
the problem will that we can have concurrent client initialisation in this case.